snowdialect.py

```diff
@@ -227,7 +228,7 @@ def _denormalize_quote_join(self, *idents):
 
     @reflection.cache
     def _current_database_schema(self, connection, **kw):
-        con = connection.connect().connection
+        con = connection.connection
```

Comment:
I'm sorry, but this line's change doesn't pass our automated tests.
We have a testcase where this gets called where no connection is open yet.

I checked https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.connect and using it here seems correct since it will reuse a previously created connection if it's available. Do you think that it was wrongly used here?

Reply:
I think this change comes from me caring about [Connection.connect](https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Connection.connect) instead of Engine.connect, but even the code change there is fine.

I'm going to rewrite this section so that it makes use of the existing `connection.connect().connection` process but uses the `Connection` context manager ensure the branched `Connection` is properly closed (which doesn't actually do anything in real terms, just ensures cleanup happens ASAP rather than relying on the GC to remove the dangling connection)

Reply:
Change pushed, hope that's suitable :)

Reply:
Sounds good! I'll kick off tests 🚀